scipy/optimize/_direct.py

```diff
@@ -0,0 +1,203 @@
+    ("The measure of the hyper-rectangle with best function value found "
+     "is below sigma_per={}"),
+)
+
+
+def direct(
+    func: Callable[[npt.ArrayLike, Tuple[Any]], float],
+    bounds: Union[Iterable, Bounds],
+    *,
+    args: tuple = (),
+    disp: bool = False,
+    iatol: float = 1e-4,
+    maxfun: int = 20000,
+    maxiter: int = 6000,
+    locally_biased: bool = True,
+    f_min: float = -np.inf,
+    f_min_per: float = 0.01,
+    vol_per: float = -1.0,
+    sigma_per: float = -1.0,
+    callback: Optional[Callable[[npt.ArrayLike], NoneType]] = None
+) -> OptimizeResult:
+    r"""
+    Solve an optimization problem using the DIRECT
+    (Dividing Rectangles) algorithm.
+
+    The algorithm is due to Jones et al. [1]_.
+
+    Parameters
+    ----------
+    func: callable
+        The objective function to be minimized.
+        ``func(x, \*args) -> float``
+        where ``x`` is an 1-D array with shape (n,) and args is a tuple of
+        the fixed parameters needed to completely specify the function.
+    bounds : sequence or `Bounds`
+        Bounds for variables. There are two ways to specify the bounds:
+        1. Instance of `Bounds` class.
+        2. ``(min, max)`` pairs for each element in ``x``, defining the finite
+        lower and upper bounds for the optimizing argument of `func`. It is
+        required to have ``len(bounds) == len(x)``. ``len(bounds)`` is used
+        to determine the number of parameters in ``x``.
+    args : tuple, optional
+        Any additional fixed parameters needed to
+        completely specify the objective function.
+    disp : bool, optional
+        Prints the evaluated `func` at every iteration.
+    iatol : float, optional
+        Ensures sufficient decrease in function value when a new potentially
+        optimal hyperrectangle is chosen. Default is 0.0001.
+    maxfun : int, optional
+        Approximate upper bound on objective function evaluations.
+        Default is 2000.
+    maxiter : int, optional
+        Maximum number of iterations. Default is 6000.
+    locally_biased : bool, optional
+        Whether to use the original [1]_ or modified [2]_ DIRECT algorithm.
+        Default is True. Possible values:
+
+        * ``locally_biased=False`` - use the original DIRECT algorithm
+        * ``locally_biased=True`` - use the modified DIRECT-l algorithm
+    f_min : float, optional
+        Function value of the global optimum.
+        By default it is a negative value whose absolute value is very large.
+        Set this value only if the global optimum is known.
+    f_min_per : float, optional
+        Terminate the optimization when the percent error satisfies:
+
+        .. math::
+
+            100*(f_{min} - f_{global})/\max(1, |f_{global}|) \leq f_{glper}
+
+        Default is 0.01.
+    vol_per : float, optional
+        Terminate the optimization once the volume of a hyperrectangle is less
+        than vol_per percent of the original hyper-rectangle.
```

Comment:
```suggestion
        than `vol_per` percent of the original hyper-rectangle.
```
according to [numpydoc standards](https://numpydoc.readthedocs.io/en/latest/format.html#parameters), parameters are to be enclosed in single backticks.
There may be other instances that deserve backticks elsewhere.

Still, I wasn't sure initially whether the number should be between 0 and 1 or 0 and 100 since the default was -1. (Now it looks like 0 and 1 based on the default above.) It might be less ambiguous as:

> Optimization terminates once the ratio between the volume of a hyper-rectangle and
the volume of the original hyper-rectangle falls below `vol_per`. Default is -1.0, indicating
that optimization will never terminate due to this condition.

(Presumably that is the meaning of -1?)